components/omnibox/browser/brave_search_provider_unittest.cc

```diff
@@ -0,0 +1,226 @@
+    TemplateURLID default_provider_id = default_t_url_->id();
+    ASSERT_NE(0, default_provider_id);
+
+    // Keywords are updated by the InMemoryHistoryBackend only after the message
+    // has been processed on the history thread. Block until history processes
+    // all requests to ensure the InMemoryDatabase is the state we expect it.
+    profile_->BlockUntilHistoryProcessesPendingRequests();
+
+    AutocompleteClassifierFactory::GetInstance()->SetTestingFactoryAndUse(
+        profile_.get(),
+        base::BindRepeating(&AutocompleteClassifierFactory::BuildInstanceFor));
+
+    client_ =
+        std::make_unique<ChromeAutocompleteProviderClient>(profile_.get());
+    provider_ = new BraveSearchProvider(client_.get(), nullptr);
+  }
+
+  void TearDown() override {
+    base::RunLoop().RunUntilIdle();
+
+    // Shutdown the provider before the profile.
+    provider_ = nullptr;
+  }
+
+ protected:
+  // Adds a search for |term|, using the engine |t_url| to the history, and
+  // returns the URL for that search.
+  GURL AddSearchToHistory(TemplateURL* t_url,
+                          std::u16string term,
+                          int visit_count) {
+    history::HistoryService* history = HistoryServiceFactory::GetForProfile(
+        profile_.get(), ServiceAccessType::EXPLICIT_ACCESS);
+    GURL search(t_url->url_ref().ReplaceSearchTerms(
+        TemplateURLRef::SearchTermsArgs(term),
+        TemplateURLServiceFactory::GetForProfile(profile_.get())
+            ->search_terms_data()));
+    static base::Time last_added_time;
+    last_added_time =
+        std::max(base::Time::Now(), last_added_time + base::Microseconds(1));
+    history->AddPageWithDetails(search, std::u16string(), visit_count,
+                                visit_count, last_added_time, false,
+                                history::SOURCE_BROWSED);
+    history->SetKeywordSearchTermsForURL(search, t_url->id(), term);
+    return search;
+  }
+
+  // Looks for a match in |provider_| with destination |url|.  Sets |match| to
+  // it if found.  Returns whether |match| was set.
+  bool FindMatchWithDestination(const GURL& url, AutocompleteMatch* match) {
+    for (auto i = provider_->matches().begin(); i != provider_->matches().end();
+         ++i) {
+      if (i->destination_url == url) {
+        *match = *i;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void QueryForInputAndSetWYTMatch(const std::u16string& text,
+                                   AutocompleteMatch* wyt_match) {
+    AutocompleteInput input(text, metrics::OmniboxEventProto::OTHER,
+                            ChromeAutocompleteSchemeClassifier(profile_.get()));
+    provider_->Start(input, false);
+
+    // RunUntilIdle so that the task scheduled by SearchProvider to create the
+    // URLFetchers runs.
+    base::RunLoop().RunUntilIdle();
+
+    profile_->BlockUntilHistoryProcessesPendingRequests();
+    if (!wyt_match)
```

Comment:
why return? `DCHECK(wyt_match);` seems more appropriate.

Reply:
Sorry, again, it was copied from the chromium tests and I didn't update this.